src/feature/featuremodel.cpp

```diff
@@ -157,144 +157,110 @@ QObject* FeatureModel::get(const QString& feature) {
+    logger.error() << "Error in the json format: experimentalFeatures is"
+                      "not an object.";
     return;
   }
 
-  QJsonArray allowParametersArray = allowParameterValue.toArray();
-  for (const QJsonValue& param : allowParametersArray) {
-    if (!param.isString()) {
-      logger.error()
-          << "Error in the json format; allowlist parameter is not a string";
+  QJsonObject experimentalFeaturesObj = experimentalFeatures.toObject();
+  logger.debug() << experimentalFeaturesObj.keys();
+  for (const QString& key : experimentalFeaturesObj.keys()) {
+    const Feature* experimentalFeature = Feature::getOrNull(key);
+    if (!experimentalFeature) {
+      logger.warning() << "Got" << key
+                       << "but experimental feature doesn't exist. Ignoring.";
       continue;
     }
-    AdjustFiltering::instance()->allowField(param.toString());
-  }
 
-  QJsonValue denyParameterValue = adjustFieldsValue["denyParameters"];
-  if (!denyParameterValue.isObject()) {
-    logger.error()
-        << "Error in the json format; deny parameters in not an object";
-    return;
-  }
-
-  QJsonObject denyParameterObject = denyParameterValue.toObject();
-  for (const QString& key : denyParameterObject.keys()) {
-    QJsonValue value = denyParameterObject.value(key);
-    if (!value.isString()) {
-      logger.error()
-          << "Error in the json format; deny list parameter is not a string";
-      continue;
+    QJsonValue experimentalFeatureSettings = experimentalFeaturesObj[key];
+    if (!experimentalFeatureSettings.isObject()) {
+      logger.error() << "Error in the json format: experimentalFeature" << key
+                     << "is not an object.";
+      return;
     }
 
-    AdjustFiltering::instance()->denyField(key, value.toString());
-  }
+    QJsonObject experimentalFeatureSettingsObj =
+        experimentalFeatureSettings.toObject();
+    for (const QString& settingKey : experimentalFeatureSettingsObj.keys()) {
+      auto value = experimentalFeatureSettingsObj[settingKey].toVariant();
+      if (value.isNull()) {
+        logger.warning()
+            << "Received null value for experimental feature setting"
+            << settingKey;
+        continue;
+      }
 
-  QJsonValue mirrorParameterValue = adjustFieldsValue["mirrorParameters"];
-  if (!mirrorParameterValue.isObject()) {
-    logger.error()
-        << "Error in the json format; mirror parameters are not an object";
-    return;
-  }
+      logger.debug() << "Setting experimental feature setting" << settingKey;
 
-  QJsonObject mirrorParameterObject = mirrorParameterValue.toObject();
-  for (const QString& key : mirrorParameterObject.keys()) {
-    QJsonValue values = mirrorParameterObject.value(key);
-    if (!values.isArray()) {
-      logger.error() << "Error in the json format; mirror parameters value is "
-                        "not an array";
-      continue;
+      experimentalFeature->settingGroup()->set(settingKey, value);
```

Comment:
Could you please point me to where this settingsGroup gets set in this PR?

Reply:
https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8904/files#diff-db1608e2b10ad038249f1acfb53d3cdb5d5ea47f624d4a1612ac4baa4bc68229R41

Reply:
Are there any uses of EXPERIMENTAL_FEATURE yet, other than in tests? 
Also, is it weird that in  `SettingGroup* m_##experimentId = `  the SettingGroup is called experimentId, or am I missing something?

Reply:
That is correct, the only experimental feature we currently have is the testing one.

Note the `##` before experimentId. That is grabbing the macro argument. Macros are confusing, sorry. Maybe it's confusing because I refer to the argument as `id` in the feature file and as `experimentId` in the settingsholder file.

Reply:
I meant, why is `experimentId` part of the name, when it actually holds a SettingsGroup?

Reply:
Well, this is an API inside the SettingsHolder, my thinking is that it would be clear this is holding the settings of an experiment because of the nature of the object holding this property. Do you have another naming suggestion?

Reply:
Are the following better?

```
SettingGroup* experimentSettings() { return m_##ExperimentSettings; }

SettingGroup* m_##ExperimentSettings =                                            \
      SettingsManager::instance()->createSettingGroup(                        \
          QString("%1/%2").arg(EXPERIMENTS_SETTING_GROUP).arg(#experimentId), \
          true, false, experimentSettings);
```